### ISSUE TITLE
Search engine results timeout

### ISSUE THREAD

**Issue by pati-ni** [searx/searx#3042]
In a self hosted setup of searx recently I get timeouts for almost all engines.

> Error! Engines cannot retrieve results.
> google (timeout), wikidata (timeout), bing (timeout), brave (timeout), wikipedia (timeout)
> Please, try again later or find another searx instance. (Public instances)

Is it a searx flaw or the search engines drop the searx queries? Any workarounds available?

I can say that this is consistent across many engines. If I disable everything except duckduckgo then I get results from duckduckgo. However, if I enable other engines I don't get any output whatsoever. Is there a timeout that invalidates the results for all the engines? Duckduckgo does not timeout but searx is not returning results for this either.

If it helps, I access searx using nginx as a reverse proxy.

**Comment by unixfox** [comment]
Might be related to https://github.com/searxng/searxng/issues/334

**Comment by pati-ni** [comment]
@unixfox Thanks for the pointer. The issue you reference is still open with no apparent solution. I tried your suggestion `enable_http2: False` but the second query gave me the same behavior.

**Comment by allendema** [comment]
Have you tried this? https://github.com/searxng/searxng/issues/334#issuecomment-922730089

You just have to use ```<15```  (or similiar) as prefix for 15 second timeout.

```<15 fediverse```

**Comment by pati-ni** [comment]
@allendema 
I have tried increasing the timeout, but it does not address the issue. It is my impression is that your suggestion will make the query time longer? 

Over the past 6 months I am using searx it is becoming progressively unreliable, not sure if search engines are throttling queries or the searx code base becomes more inefficient and demands more resources.

At the moment, it is not in a usable state and I hope there is a way to address or mitigate this problem.

**Comment by unixfox** [comment]
Do you experience this issue directly after restarting searx or when searx has been running for a long time?

**Comment by pati-ni** [comment]
I restart searx through uwsgi each time I update from github. I can recall a slow first query after a restart but I do not think that over time the problem improves. The instance runs 24/7 on a VPS

**Comment by unixfox** [comment]
I'm not talking about the problem improving over time, but if you can reproduce the issue directly after restarting or if it happens later when searx has been running for a long time.

**Comment by Yuerchu** [comment]
我也遇到了一样的情况，即便我把它设置成60秒的等待时间，依旧没有解决。

**Comment by victoriastuart** [comment]
Over recent months this issue effectively killed the use of `searx` for me (installed locally on localhost).

```html
Error! Engines cannot retrieve results.
google (timeout), wikidata (timeout), bing (timeout), brave (timeout), wikipedia (timeout)
Please, try again later or find another searx instance. (Public instances)
```

It appears to be exacerbated by background internet activity (file downloads), etc.

After numerous internet searches and `searx` configuration tweaks, I reluctantly started searching for alternative metasearch options, finding a reddit post that recommended `SearXNG`.

I installed that as described here - adopting the easy route: using my Arch Linux package manager (not the horrendous manual installation, which fails anyway due to bugs in the instructions),

https://github.com/searxng/searxng/issues/1168

and after using `SearXNG` for ~8h now, it has been a breath of fresh air - executing flawlessly (no timeouts; ...).

---

**Postscript.** No longer the case: SearXNG also problematic; similar issues.

**Comment by elvismercado** [comment]
So has there been any progress with this issue?

I havce been trying to install searx or searxng via docker on my unraid server.

But always i am getting the same error described here.

Raising the timeout helps get in some results but in my case they are not even cached so each and every single search takes the same amount of time as set in the timeout.

**Comment by unixfox** [comment]
> So has there been any progress with this issue?
> 
> I havce been trying to install searx or searxng via docker on my unraid server.
> 
> But always i am getting the same error described here.
> 
> Raising the timeout helps get in some results but in my case they are not even cached so each and every single search takes the same amount of time as set in the timeout.

Did you update SearX to the latest version? There was a new version: https://github.com/searx/searx/releases/tag/v1.1.0

Or you could consider switching to [SearXNG](https://github.com/searxng/searxng/) (a fork of SearX) which don't have this issue like described in https://github.com/searx/searx/issues/3042#issuecomment-1116872698

**Comment by elvismercado** [comment]
Yes I used both searx and searxng latest in docker containers and still have the same issue.
So dont think its fixed because of switching to searxng

**Comment by unixfox** [comment]
How did you install Searx?

Which engines are you using? Is it the default settings?

**Comment by elvismercado** [comment]
I installed using searxng docker image. Default settings.
I even removed all the engines and only left the google engine to test

**Comment by unixfox** [comment]
Can you join the IRC or matrix channel to discuss this issue "more freely": https://github.com/searxng/searxng#contact
Just ping me when you are there

**Comment by imacmill** [comment]
> Can you join the IRC or matrix channel to discuss this issue "more freely": https://github.com/searxng/searxng#contact Just ping me when you are there

Did anything ever come from your 'more free' conversation? I'm having the exact same issue as the OP on a privately hosted instance of SearXNG. It's installed on a NUC6CAY running DietPi. It's a reasonably capable little box. I get constant timeouts on all configured engines. If I keep clicking the search button, I'll randomly get more or fewer timeouts each time.

I've read quite a few posts about this same issue with Searx, but all of them are a couple of years old, so I don't know SearX's status as of today.

**Comment by elvismercado** [comment]
I haven't been able to pick this up again because it's very time consuming. Been busy with some other stuff.
But will try to pick this up soon. Hopefully @unixfox has some insights for us.

**Comment by xtay573269555** [comment]
I got the Google timeout error. The other search engines is ok. I turn off http2 [https://github.com/searxng/searxng/issues/334#issuecomment-940070743](https://github.com/searxng/searxng/issues/334#issuecomment-940070743 ), and then it works.
